flake8_trio.py

```diff
@@ -472,108 +474,235 @@ def visit_Call(self, node: ast.Call):
         self.generic_visit(node)
 
 
-class Visitor107_108(Flake8TrioVisitor):
+class Visitor107_108_109(Flake8TrioVisitor):
     def __init__(self) -> None:
         super().__init__()
         self.all_await = True
+        self.checkpoint_after_yield = True
+        self.has_yield = False
+        self.checkpoint_continue = True
+        self.checkpoint_break = True
+
+    def get_state(self) -> Tuple[bool, bool, bool, bool]:
+        return (
+            self.all_await,
+            self.checkpoint_after_yield,
+            self.has_yield,
+            self.suppress_errors,
+        )
+
+    def set_state(self, state: Tuple[bool, bool, bool, bool]):
+        (
+            self.all_await,
+            self.checkpoint_after_yield,
+            self.has_yield,
+            self.suppress_errors,
+        ) = state
 
     def visit_AsyncFunctionDef(self, node: ast.AsyncFunctionDef):
-        outer = self.all_await
+        outer = self.get_state()
 
         # do not require checkpointing if overloading
         self.all_await = has_decorator(node.decorator_list, "overload")
```

Comment:
What's the guarantees on `[async]contextmanager`s? Checkpoint on enter and exit, or at least one of enter and exit?

Reply:
At least one of enter and exit.